### DISCUSSION TITLE
High CPU, HW accel not working?

### DISCUSSION THREAD

**Discussion by bruceda-nz** [roflcoopter/viseron#818] [Q&A] [answered]
Hi all

I have Viseron up and running on an old Ubuntu machine but I am struggling with really high CPU use. The machine has an i5-3470 which should support QSV and definitely supports VAAPI. 

I am exposing /dev/dri in my docker-compose file 

when I bring the container up, it runs but I get an error libva: /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so init failed

If I execute vainfo in the container it says similar about iHD and then seems to fall back to i965 and then work.

I have one camera running currently for testing and even if I use the low res substream, when I run top in the host OS it shows python consuming  300+% CPU time. The same thing happens regardless of stream.

intel_gpu_top in the host OS shows activity but it's the CPU usage that has me puzzled.

I have set the detector to motion only to see if it creates peaks but once the stream starts the CPU goes high.

I know the CPU is old and not that capable but I thought with a 640x480 substream at 5fps I'd be ok.

Any ideas of where to start looking?

**Comment by roflcoopter** [comment]
If you run `top -c` inside the container, what is the command behind the process that is taking all the CPU?

**Reply by bruceda-nz** [reply]
viseron -u -m viseron

Is that what you were expecting?

**Reply by roflcoopter** [reply]
Strange that the Viseron process is using so much CPU. It only runs image decoding and optionally motion detection which shouldnt be that demanding.

Can you show me your config?

**Reply by bruceda-nz** [reply, marked as answer]
I have just sorted it out now.

Entirely my own fault, I had object detection on and set to detect on motion but then no motion detection enabled.

**Reply by roflcoopter** [reply]
Great!

**Comment by bruceda-nz** [comment]
Bump

@roflcoopter any thoughts on this?
